src/compat/posix/sys/mman/mprotect.c

```diff
@@ -0,0 +1,49 @@
+/**
+ * @file mprotect.c
+ * @brief Set protection of memory mapping
+ * @author Archit Checker <archit.checker5@gmail.com>
+ * @version
+ * @date 24.03.2020
+ */
+
+#include <errno.h>
+#include <stddef.h>
+
+#include <sys/mman.h>
+#include <hal/mmu.h>
+#include <mem/vmem.h>
+#include <mem/mmap.h>
+#include <kernel/task/resource/mmap.h>
+
+int mprotect(void* addr, size_t len, int prot) {
+	int err;
+	int curr_prot = PROT_NONE;
+	struct emmap *emmap = task_self_resource_mmap();
+
+	if ((int) addr % VMEM_PAGE_SIZE != 0) {
+		err = EINVAL;
+		return SET_ERRNO(err);
+	}
+
+	if (prot & PROT_READ) {
+		curr_prot |= PROT_READ;
+	}
+
+	if (prot & PROT_WRITE) {
+		curr_prot |= PROT_WRITE;
+	}
+
+	if (prot & PROT_EXEC) {
+		curr_prot |= PROT_EXEC;
+	}
```

Comment:
Can we just:

```
cur_prot |= prot & (PROT_READ | PROT_WRITE | PROT_EXEC);
```

Moreover, why do you need `cur_prot` if you can work directly with `prot`? You can just clear all bits except PROT_READ/WRITE/EXEC and pass it further.

Reply:
Yes, that is correct. I've fixed it in the latest fixup.